reframe/core/schedulers/torque.py

```diff
@@ -0,0 +1,117 @@
+import re
+
+from reframe.core.config import settings
+from reframe.core.exceptions import JobError
+from reframe.core.logging import getlogger
+from reframe.core.schedulers.pbs import PbsJobScheduler, _run_strict
+from reframe.core.schedulers.registry import register_scheduler
+import reframe.utility.os_ext as os_ext
+
+
+JOB_STATES = {
+    'Q': 'QUEUED',
+    'H': 'HELD',
+    'R': 'RUNNING',
+    'E': 'EXITING',
+    'T': 'MOVED',
+    'W': 'WAITING',
+    'S': 'SUSPENDED',
+    'C': 'COMPLETED',
+}
+
+
+@register_scheduler('torque')
+class TorqueJobScheduler(PbsJobScheduler):
+    def _update_state(self, job):
+        '''Check the status of the job.'''
+
+        completed = os_ext.run_command('qstat -f %s' % job.jobid, log=False)
+
+        # Needed for clusters that have the keep_completed parameter set to 0
+        if completed.returncode == 153:
+            getlogger().debug(
+                'jobid not known by scheduler, assuming job completed'
+            )
+            job.state = 'COMPLETED'
+            return
+
+        if completed.returncode != 0:
+            raise JobError(completed.returncode, completed.stderr)
+
+        state_match = re.search(
+            r'^\s*job_state = (?P<state>[A-Z])', completed.stdout, re.MULTILINE
+        )
+        if not state_match:
+            getlogger().debug(
+                'job state not found (stdout follows)\n%s' % completed.stdout
+            )
+            return
+
+        state = state_match.group('state')
+        job.state = JOB_STATES[state]
+
+        if job.state == 'COMPLETED':
+            code_match = re.search(
+                r'^\s*exit_status = (?P<code>\d+)',
+                completed.stdout,
+                re.MULTILINE,
+            )
+            if not code_match:
+                return
+            job.exitcode = int(code_match.group('code'))
+
+    def _emit_lselect_option(self, job):
+        num_tasks_per_node = job.num_tasks_per_node or 1
+        num_cpus_per_task = job.num_cpus_per_task or 1
+        num_nodes = job.num_tasks // num_tasks_per_node
+        num_cpus_per_node = num_tasks_per_node * num_cpus_per_task
+
+        select_opt = '-l nodes=%s:ppn=%s' % (num_nodes, num_cpus_per_node)
```

Comment:
I think that this is the only difference with the PBS `_emit_lselect_option`, right? We should find then a better way to avoid all this code duplication.

Reply:
ok, I'll try to factor it out.